### PR TITLE
Doesn't do `run_query` when `compile`

### LINKED ISSUE #215: `dbt compile` fails on versions `> 0.1.1`
### Describe the bug
On `dbt 1.2.1`, and `dbt_project_evaluator > 0.1.1` the `dbt compile` command fails due to the `dbt_project_evaluator_exceptions` model being missing

### Steps to reproduce
1. Edit the `dbt-labs/dbt_project_evaluator` to the appropriate version (I tested on `0.1.1`, `0.1.2`, `0.1.3`, and `0.1.4`)
2. Run `dbt clean && dbt deps && dbt compile`

### Expected results
Successful compilation of dbt models.

### Actual results
The compile command fails due to a missing `dbt_project_evaluator_exceptions` model

### Screenshots and log output
using version `0.1.4`:
```
$ dbt clean && dbt deps && dbt compile
19:14:17  Running with dbt=1.2.1
19:14:17  Checking target/*
19:14:17  Cleaned target/*
19:14:17  Checking dbt_packages/*
19:14:18  Cleaned dbt_packages/*
19:14:18  Finished cleaning all paths.
19:14:21  Running with dbt=1.2.1
19:14:22  Installing dbt-labs/dbt_utils
19:14:23    Installed from version 0.8.6
19:14:23    Updated version available: 0.9.2
19:14:23  Installing calogica/dbt_expectations
19:14:25    Installed from version 0.5.8
19:14:25    Updated version available: 0.6.1
19:14:25  Installing dbt-labs/metrics
19:14:26    Installed from version 0.3.4
19:14:26    Up to date!
19:14:26  Installing dbt-labs/codegen
19:14:26    Installed from version 0.8.0
19:14:26    Up to date!
19:14:26  Installing data-mie/dbt_profiler
19:14:26    Installed from version 0.4.1
19:14:26    Up to date!
19:14:26  Installing dbt-labs/dbt_project_evaluator
19:14:27    Installed from version 0.1.4
19:14:27    Up to date!
19:14:27  Installing calogica/dbt_date
19:14:28    Installed from version 0.5.7
19:14:28    Updated version available: 0.6.0
19:14:28  
19:14:28  Updates available for packages: ['dbt-labs/dbt_utils', 'calogica/dbt_expectations', 'calogica/dbt_date']                 
Update your versions in packages.yml, then run dbt deps
19:14:31  Running with dbt=1.2.1
19:14:31  Partial parse save file not found. Starting full parse.
19:14:35  Found 196 models, 80 tests, 2 snapshots, 0 analyses, 869 macros, 0 operations, 10 seed files, 76 sources, 2 exposures, 21 metrics
19:14:35  
19:14:36  Concurrency: 4 threads (target='dev')
19:14:36  
19:14:41  Encountered an error:
Runtime Error
  Runtime Error in model fct_source_directories (models/marts/structure/fct_source_directories.sql)
    404 Not found: Table MY-GCS-PROJECT:MY_DBT_DATASET.dbt_project_evaluator_exceptions was not found in location US
    
    (job ID: MY-BIGQUERY-JOB-ID)
19:14:41  Traceback (most recent call last):
  File "/workspaces/inquirer-dbt/.venv/lib/python3.9/site-packages/dbt/main.py", line 129, in main
    results, succeeded = handle_and_check(args)
  File "/workspaces/inquirer-dbt/.venv/lib/python3.9/site-packages/dbt/main.py", line 191, in handle_and_check
    task, res = run_from_args(parsed)
  File "/workspaces/inquirer-dbt/.venv/lib/python3.9/site-packages/dbt/main.py", line 238, in run_from_args
    results = task.run()
  File "/workspaces/inquirer-dbt/.venv/lib/python3.9/site-packages/dbt/task/runnable.py", line 470, in run
    result = self.execute_with_hooks(selected_uids)
  File "/workspaces/inquirer-dbt/.venv/lib/python3.9/site-packages/dbt/task/runnable.py", line 433, in execute_with_hooks
    res = self.execute_nodes()
  File "/workspaces/inquirer-dbt/.venv/lib/python3.9/site-packages/dbt/task/runnable.py", line 379, in execute_nodes
    self.run_queue(pool)
  File "/workspaces/inquirer-dbt/.venv/lib/python3.9/site-packages/dbt/task/runnable.py", line 293, in run_queue
    self._raise_set_error()
  File "/workspaces/inquirer-dbt/.venv/lib/python3.9/site-packages/dbt/task/runnable.py", line 274, in _raise_set_error
    raise self._raise_next_tick
dbt.exceptions.RuntimeException: Runtime Error
  Runtime Error in model fct_source_directories (models/marts/structure/fct_source_directories.sql)
    404 Not found: Table MY-GCS-PROJECT:MY_DBT_DATASET.dbt_project_evaluator_exceptions was not found in location US
    
    (job ID: 9372cd43-37cd-4636-a7ac-c4e30b4ca022)

```

### System information
**The contents of your `packages.yml` file:**
```yaml
packages:
  - package: dbt-labs/dbt_utils
    version: [">=0.8.6", "<0.10.0"]
  - package: calogica/dbt_expectations
    version: [">=0.5.8", "<0.6.0"]
  - package: dbt-labs/metrics
    version: [">=0.3.1","<0.4.0"]
  - package: dbt-labs/codegen
    version: 0.8.0
  - package: data-mie/dbt_profiler
    version: 0.4.1
  - package: dbt-labs/dbt_project_evaluator
    version: 0.1.4

```

**Which database are you using dbt with?**
- [ ] postgres
- [ ] redshift
- [x] bigquery
- [ ] snowflake
- [ ] other (specify: ____________)

**The output of `dbt --version`:**
```
$ dbt --version
Core:
  - installed: 1.2.1
  - latest:    1.2.1 - Up to date!

Plugins:
  - bigquery: 1.2.0 - Up to date!
```

### Additional context
Note that I get the expected behavior on version `0.1.1`

### Are you interested in contributing the fix?
Willing to help out, but would need a hand getting started.

### PR DESCRIPTION
**Pull request by b-per**
This is a:
- [x] bug fix PR with no breaking changes
- [ ] new functionality

## Link to Issue 

Closes #215

### CODE REVIEW

File: macros/filter_exceptions.sql [hunk 1]
```diff
@@ -8,7 +8,7 @@ from {{ ref('dbt_project_evaluator_exceptions') }}
 where fct_name = '{{ model_ref.name }}'
 {% endset %}
 
-{% if execute %}
+{% if execute and flags.WHICH not in ['compile'] %}
```
Comment (dave-connors-3) [inline macros/filter_exceptions.sql:11]: may want to add `rpc` to this list too!

**Review by dave-connors-3 — APPROVED** [review]
🦞

**Comment by dave-connors-3** [comment]
@b-per this looks good to be once we have an understanding of how we want the IDE to behave in this case (also is rpc going away in the newer iterations of the IDE?)

**Comment by b-per** [comment]
I believe that `WHICH` shows as `rpc` only when clicking on preview/compile. 

I'd rather not alter the behavior if people want to see the query being run (and want to see the true results taking in account their exceptions). They will have an error if they try to compile/preview without having loaded the seed but I'd actually be ok with that.

**Review by graciegoheen — APPROVED** [review]
LGTM :)

**Pull request merged**